pkgs/development/python-modules/jpylyzer/default.nix

```diff
@@ -33,19 +37,23 @@ in buildPythonPackage rec {
 
   checkInputs = [ pytestCheckHook lxml ];
 
-  # don't depend on testFiles on darwin as it may not be extractable due to
-  # weird filenames
-  preCheck = lib.optionalString (!stdenv.isDarwin) ''
+  # don't depend on testFiles unless doFullCheck as it may not be extractable
+  # on some filesystems due to weird filenames
+  preCheck = lib.optionalString doFullCheck ''
     sed -i '/^testFilesDir = /ctestFilesDir = "${testFiles}"' tests/unit/test_testfiles.py
   '';
-  disabledTestPaths = lib.optionals stdenv.isDarwin [
+  disabledTestPaths = lib.optionals (!doFullCheck) [
     "tests/unit/test_testfiles.py"
   ];
 
   pythonImportsCheck = [ "jpylyzer" ];
 
   disallowedReferences = [ testFiles ];
 
+  passthru.tests = {
+    withFullCheck = jpylyzer.override { doFullCheck = true; };
+  };
+
```

Comment:
I am not sure we even need this. I would be fine with just disabling the test

Reply:
I'd rather like the ability to perform extended tests because the remaining tests don't actually exercise that much.